http/handler.go

```diff
@@ -731,14 +731,10 @@ func parseJSONRequest(r *http.Request, w http.ResponseWriter, out interface{}) (
 	// Limit the maximum number of bytes to MaxRequestSize to protect
 	// against an indefinite amount of data being read.
 	reader := r.Body
-	var origBody io.ReadWriter
```

Comment:
Is this a problem? The comment above seems to be ignored here.

Reply:
@phyrog I think this is the only question I have -- I'm happy splitting to a separate issue after investigation but should probably understand it before we do the removal and merge this one.

Reply:
I guess we can just wrap body in an io.LimitedReader to comply with the comment. But I think the removal here is unrelated to the comment as it is now (maybe there used to be more code that handled that there).

Reply:
@cipherboy Request size limit should be handled by the `wrapMaxRequestSizeHandler` middleware, though I have not checked yet if this is called before `parseJSONRequest` in all cases, or if there are edge cases where it is not (recovery mode maybe?).